Source/DafnyCore/Compilers/Compiler-go.cs

```diff
@@ -3648,18 +3648,7 @@ private class SpecialNativeType : UserDefinedType {
         psi.EnvironmentVariables["GOTMPDIR"] = localAppData + @"\Temp";
         psi.EnvironmentVariables["LOCALAPPDATA"] = localAppData + @"\go-build";
       }
-
-      try {
-        using var process = Process.Start(psi);
-        if (process == null) {
-          return false;
-        }
-        process.WaitForExit();
-        return process.ExitCode == 0;
-      } catch (System.ComponentModel.Win32Exception e) {
-        outputWriter.WriteLine("Error: Unable to start go ({0}): {1}", psi.FileName, e.Message);
-        return false;
-      }
+      return RunProcess(psi, null, "go", outputWriter) == 0;
```

Comment:
Did you mean 
```suggestion
      return 0 == RunProcess(psi, null, "go", outputWriter);
```
?
But other than that, since you are never using the "integer" part of RunProcess, why don't you have `RunProcess` return `true` if it succeeded, and `false` otherwise? That way you don't need an extra superfluous equality.

Reply:
Because I expect that eventually we will care about the return code from the target program and RunProcess will propagate it so `dafny run` can return it (or at least report it). However, if you insist, I can push that equality check into RunProcess and out of the individual compilers.

Reply:
No I won't insist, I think you have a point.